client/components/social-buttons/google.js

```diff
@@ -194,6 +172,53 @@ class GoogleSocialButton extends Component {
 		this.props.responseHandler( { access_token, id_token } );
 	}
 
+	async fetchNonceAndInitializeGoogleSignIn() {
+		try {
+			const response = await wpcomRequest( {
+				path: '/generate-authorization-nonce',
+				apiNamespace: 'wpcom/v2',
+				apiVersion: '2',
+				method: 'GET',
+			} );
+			const state = response.nonce;
+
+			await this.initializeGoogleSignIn( state );
+		} catch ( error ) {
+			this.props.showErrorNotice(
+				this.props.translate(
+					'Error fetching nonce or initializing Google sign-in. Please try again.'
+				)
+			);
+		}
+	}
+
+	async handleClickNew( event ) {
+		event.preventDefault();
+		event.stopPropagation();
+
+		if ( this.state.error && this.state.eventTimeStamp !== event.timeStamp ) {
+			this.setState( {
+				showError: ! this.state.showError,
```

Comment:
Is this correct that we are toggling the `this.state.showError`? I think it would be more appropriate to set the `showError` to `true`.  Also, is this the expected behavior that in a case where the `handleClickNew` is called and we have the `this.state.error` we will still make the calls: 
```
await this.fetchNonceAndInitializeGoogleSignIn();

this.props.onClick( event );
```

Reply:
Just to give a quick update, me and @ivan-ottinger  had a look at the function on call. It seems like the:
```
if ( this.state.error ) {
		return;
}
```
check after the `await this.fetchNonceAndInitializeGoogleSignIn();` call is useful, as it may set the `error` state var, when the google API fails to be loaded. However, the `if ( this.state.error && this.state.eventTimeStamp !== event.timeStamp )` check is unnecessary. as the `handleClickNew` will never be executed when `this.state.error` is present, as it disables the button.